third_party/java/arcs/build_defs/internal/kotlin.bzl

```diff
@@ -24,19 +24,21 @@ def arcs_kt_binary(name, srcs = [], deps = [], visibility = None):
     """Performs final compilation of wasm and bundling if necessary."""
     libname = name + "_lib"
 
-    # Declare a library because g3 kt_native_binary doesn't take srcs
-    kt_native_library(
-        name = libname,
-        srcs = srcs,
-        deps = _ARCS_KOTLIN_LIBS + deps,
-        tags = ["wasm"],
-        visibility = visibility
-    )
+    if srcs:
+        # Declare a library because g3 kt_native_binary doesn't take srcs
+        kt_native_library(
+            name = libname,
+            srcs = srcs,
+            deps = _ARCS_KOTLIN_LIBS + deps,
+            visibility = visibility
+        )
+
+        deps = [":%s" % libname] + deps
```

Comment:
I think you can just set `deps  = [":" + libname]` here. The binary depends only on the library; the library depends on everything else.

Reply:
I think I tried that before and there was some strict_deps issue, im ok with removing it, but test internally first.

Reply:
Done

Reply:
Done